qiskit/transpiler/runningpassmanager.py

```diff
@@ -118,6 +119,22 @@ def run(self, circuit, output_name=None, callback=None):
         else:
             circuit.name = name
         circuit._layout = self.property_set['layout']
```

Comment:
Do the same, here. Replace the below with:

```
if calibrations:
    circuit.calibrations = calibrations
```
You might just have to update `calibrations` with a setter. I don't see any problem with doing that.

Reply:
Actually, probably need:
```
if calibrations:
    circuit.calibrations = copy(calibrations)
```

Reply:
Fixed in https://github.com/Qiskit/qiskit-terra/pull/5049/commits/98ae90ffb5bf534ca41c72b1ce5ab465c7f687e3